server/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java

```diff
@@ -154,13 +154,24 @@ public TableInfo getTableInfo(String name) {
         Metadata metadata = clusterService.state().metadata();
         DocTableInfo docTableInfo = docTableByName.get(name);
         try {
+            RelationName relation = new RelationName(schemaName, name);
             if (docTableInfo == null) {
                 return docTableByName.computeIfAbsent(
                     name,
-                    n -> docTableInfoFactory.create(new RelationName(schemaName, n), metadata)
+                    n -> {
+                        return docTableInfoFactory.create(relation, metadata);
+                    }
                 );
             }
-            if (docTableInfo.metadataVersion() < metadata.version()) {
+            IndexTemplateMetadata indexTemplateMetadata = metadata.templates().get(PartitionName.templateName(schemaName, name));
```

Comment:
I'd personally extract this into its own method to make the control flow more obvious, but that's a matter of taste so entirely up to you :)

Reply:
Changed it